src/lib.rs

```diff
@@ -566,28 +573,28 @@ impl Actor {
         rt.transaction(|st: &mut State, rt| {
             if params.destination == st.network_name {
                 return Err(actor_error!(
-                illegal_argument,
-                "destination is the current network, you are better off with a good ol' message, no cross needed"
-            ));
+                    illegal_argument,
+                    "destination is the current network, you are better off with a good ol' message, no cross needed"
+                ));
             }
             // we disregard the to of the message. the caller is the one set as the from of the
             // message.
-            msg.to = match IPCAddress::new_from_ipc(&params.destination, &msg.to) {
+            msg.msg.to = match IPCAddress::new_from_ipc(&params.destination, &msg.msg.to) {
```

Comment:
`let mut msg = params.msg.clone();`, I might be wrong, but it feels like there is no need to clone, can just use sth like this: https://doc.rust-lang.org/rust-by-example/flow_control/match/destructuring/destructure_structures.html?

Reply:
I guess you put the comment to a wrong line. Yes, I would like to avoid cloning [here](https://github.com/consensus-shipyard/ipc-gateway/pull/10/commits/c3fbc75be9723a69dcfe09173a373b44a6ef2899#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R639), but I need pass serialized `params` [here](https://github.com/consensus-shipyard/ipc-gateway/pull/10/commits/c3fbc75be9723a69dcfe09173a373b44a6ef2899#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R687) and [here](https://github.com/consensus-shipyard/ipc-gateway/pull/10/commits/c3fbc75be9723a69dcfe09173a373b44a6ef2899#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R736).